pkg/operation/botanist/controlplane.go

```diff
@@ -1224,33 +1266,42 @@ func (b *Botanist) DeployETCD(ctx context.Context) error {
 	return nil
 }
 
-// CheckVPNConnection checks whether the VPN connection between the control plane and the shoot networks
+// CheckTunnelConnection checks if the tunnel connection between the control plane and the shoot networks
 // is established.
-func (b *Botanist) CheckVPNConnection(ctx context.Context, logger *logrus.Entry) (bool, error) {
+func (b *Botanist) CheckTunnelConnection(ctx context.Context, logger *logrus.Entry, tunnelName common.TunnelName) (bool, error) {
 	podList := &corev1.PodList{}
-	if err := b.K8sShootClient.Client().List(ctx, podList, client.InNamespace(metav1.NamespaceSystem), client.MatchingLabels{"app": "vpn-shoot"}); err != nil {
+	if err := b.K8sShootClient.Client().List(ctx, podList, client.InNamespace(metav1.NamespaceSystem), client.MatchingLabels{"app": string(tunnelName)}); err != nil {
 		return retry.SevereError(err)
 	}
 
-	var vpnPod *corev1.Pod
+	var tunnelPod *corev1.Pod
 	for _, pod := range podList.Items {
 		if pod.Status.Phase == corev1.PodRunning {
-			vpnPod = &pod
+			tunnelPod = &pod
 			break
 		}
 	}
 
-	if vpnPod == nil {
-		logger.Info("Waiting until a running vpn-shoot pod exists in the Shoot cluster...")
-		return retry.MinorError(fmt.Errorf("no running vpn-shoot pod found yet in the shoot cluster"))
+	if tunnelPod == nil {
+		logger.Infof("Waiting until a running %s pod  exists in the Shoot cluster...", tunnelName)
+		return retry.MinorError(fmt.Errorf("no running %s pod found yet in the shoot cluster", tunnelName))
 	}
 
-	if err := b.K8sShootClient.CheckForwardPodPort(vpnPod.Namespace, vpnPod.Name, 0, 22); err != nil {
-		logger.Info("Waiting until the VPN connection has been established...")
-		return retry.MinorError(fmt.Errorf("could not forward to vpn-shoot pod: %v", err))
+	switch tunnelName {
+	case common.VPNTunnel:
+		if err := b.K8sShootClient.CheckForwardPodPort(tunnelPod.Namespace, tunnelPod.Name, 0, 22); err != nil {
+			logger.Info("Waiting until the tunnel connection has been established...")
+			return retry.MinorError(fmt.Errorf("could not forward to %s pod: %v", common.VPNTunnel, err))
+		}
+	case common.KonnectivityTunnel:
+		logs, err := kubernetes.GetPodLogs(b.K8sShootClient.Kubernetes().CoreV1().Pods(metav1.NamespaceSystem), tunnelPod.Name, &corev1.PodLogOptions{})
```

Comment:
Why can't we keep the `CheckForwardPodPort` method here and instead fetch the logs? In the end it is kind of equivalent because both operations only work if connectivity exists, though. In what situation does the konnectivity-agent log "Invalid server"?

Reply:
what ports are we to forward, the konnectivity pod does not expose any ports.

Reply:
Hm, I think it doesn't matter as we don't send anything to this port. We just check whether the API server can open the connection, that's good enough for the check. WDYT?

Reply:
can do I guess.